packages/application-components/src/components/custom-views/custom-view-loader/custom-view-loader.tsx

```diff
@@ -49,6 +49,87 @@ const CustomPanelIframe = styled.iframe`
   border: none;
 `;
 
+function useIFrameInitializationEffect({
+  iFrameRef,
+  onInitialized,
+  maxAttempts = 10,
+  pollInterval = 100,
+}: {
+  iFrameRef: React.RefObject<HTMLIFrameElement | null>;
+  onInitialized: () => void;
+  maxAttempts?: number;
+  pollInterval?: number;
+}) {
+  const [shouldStartPolling, setShouldStartPolling] = useState(false);
+  const attemptsRef = useRef(0);
+
+  const startPolling = useCallback(() => {
+    setShouldStartPolling(true);
+  }, []);
+
+  const stopPolling = useCallback(() => {
+    setShouldStartPolling(false);
+  }, []);
+
+  useEffect(() => {
+    if (!shouldStartPolling) return;
+
+    attemptsRef.current = 0;
+    let timeoutId: NodeJS.Timeout;
+
+    const pollForAppLoader = () => {
+      const iFrameDocument = iFrameRef.current?.contentWindow?.document;
+      const appLoaderElement = iFrameDocument?.getElementById('app-loader');
+
+      attemptsRef.current++;
+
+      if (attemptsRef.current > maxAttempts) {
+        console.log(
+          '[CustomViewLoader] max polling attempts reached. Failed to initialize iFrame.'
+        );
+        stopPolling();
+        return;
+      }
+
+      if (!appLoaderElement) {
+        console.log(
+          '[CustomViewLoader] no app-loader yet, not sending initialization messages.'
+        );
+
+        return;
+      }
+
+      if (appLoaderElement) {
+        console.log(
+          '[CustomViewLoader] app-loader found, sending initialization messages.'
+        );
+        setTimeout(onInitialized, pollInterval * 5);
```

Comment:
Apparently we still need to delay sending the message. 

Assumption being that while onLoad indicates JS execution it doesn't indicate rendering and component mounting. Basically meaning the code inside the iFrame isn't ready yet.